database-plugins/src/main/java/io/cdap/plugin/DBManager.java

```
@@ -44,18 +46,24 @@ public DBManager(ConnectionConfig config) {
-    Preconditions.checkArgument(!(config.user == null && config.password != null),
-                                "user is null. Please provide both user name and password if database requires " +
-                                  "authentication. If not, please remove password and retry.");
+                                                            String jdbcPluginId, FailureCollector collector) {
+    if (config.user == null && config.password != null) {
+      collector.addFailure("User is null. Please provide both user name and password if database requires " +
+              "authentication. If not, please remove password and retry.", null)
+              .withConfigProperty(ConnectionConfig.USER).withConfigProperty(ConnectionConfig.PASSWORD);
+    }
+
     Class<? extends Driver> jdbcDriverClass = pipelineConfigurer.usePluginClass(config.jdbcPluginType,
                                                                                 config.jdbcPluginName,
                                                                                 jdbcPluginId,
                                                                                 PluginProperties.builder().build());
-    Preconditions.checkArgument(
-      jdbcDriverClass != null, "Unable to load JDBC Driver class for plugin name '%s'. Please make sure that the " +
-        "plugin '%s' of type '%s' containing the driver has been installed correctly.", config.jdbcPluginName,
-      config.jdbcPluginName, config.jdbcPluginType);
+    if (jdbcDriverClass == null) {
+      collector.addFailure(String.format("Unable to load JDBC Driver class for plugin name '%s'.",
+              config.jdbcPluginName), String.format("Please make sure that the plugin '%s' of type '%s' containing " +
+              "the driver has been installed correctly.", config.jdbcPluginName, config.jdbcPluginType))
+              .withConfigProperty(ConnectionConfig.JDBC_PLUGIN_NAME);
```

`.withPluginNotFound()`

---

This should throw an exception or null driver classes should be handled by callers and method should be marked as `@Nullable`